Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -0,0 +1,212 @@
+                IsTitleAboveName = isTitleAboveName,
+                IsTitleVisible = isTitleVisible,
+                IconID = (NameplateStatusIcons)iconId,
+            };
+
+            // Add known nameplate elements
+            nameplateInfo.Elements.AddRange([
+                new(titlePtr, NameplateElementName.Title),
+                new(namePtr, NameplateElementName.Name),
+                new(freeCompanyPtr, NameplateElementName.FreeCompany),
+                new(prefixPtr, NameplateElementName.Prefix)
+            ]);
+
+            // Create new nameplate object
+            var namePlateObj = new NameplateObject(playerNameplateObjectPtr, nameplateInfo);
+
+            // Invoke event
+            this.OnNameplateUpdate.Invoke(namePlateObj);
+
+            // Get new states
+            isTitleAboveName = nameplateInfo.IsTitleAboveName;
+            isTitleVisible = nameplateInfo.IsTitleVisible;
+            iconId = (int)nameplateInfo.IconID;
+
+            // Handle changed elements
+            foreach (var element in nameplateInfo.Elements)
+            {
+                if (element.HasChanged)
+                {
+                    // Copy back known element properties
+                    switch (element.Name)
+                    {
+                        case NameplateElementName.Title:
+                            titlePtr = this.PluginAllocate(element.Text);
+                            break;
+                        case NameplateElementName.Name:
+                            namePtr = this.PluginAllocate(element.Text);
+                            break;
+                        case NameplateElementName.FreeCompany:
+                            freeCompanyPtr = this.PluginAllocate(element.Text);
+                            break;
+                        case NameplateElementName.Prefix:
+                            prefixPtr = this.PluginAllocate(element.Text);
+                            break;
+                    }
+                }
+            }
+        }
+
+        // Call original
+        var result = this.setPlayerNameplateDetourHook.Original(
+            playerNameplateObjectPtr,
+            isTitleAboveName,
+            isTitleVisible,
+            titlePtr,
+            namePtr,
+            freeCompanyPtr,
+            prefixPtr,
+            iconId);
+
+        // Free pointers
+        if (ptrToFree.Count > 0)
+            ptrToFree.ForEach(this.PluginFree);
+
+        // Return result
+        return result;
+    }
+
+    private nint PluginAllocate(SeString seString)
+    {
+        return this.PluginAllocate(seString.Encode());
+    }
+
+    private nint PluginAllocate(byte[] bytes)
+    {
+        var pointer = Marshal.AllocHGlobal(bytes.Length + 1);
+        Marshal.Copy(bytes, 0, pointer, bytes.Length);
+        Marshal.WriteByte(pointer, bytes.Length, 0);
+        return pointer;
+    }
+
+    private void PluginFree(nint ptr)
+    {
+        Marshal.FreeHGlobal(ptr);
+    }
```

Comment:
I feel like there has to be a better way, but we can look into that later, as this is implementation details that we can address at any time without breaking the API.